src/compiler/parser.ts

```diff
@@ -1010,6 +1013,14 @@ module ts {
             setContextFlag(val, ParserContextFlags.GeneratorParameter);
         }
 
+        function setAwaitContext(val: boolean) {
+            setContextFlag(val, ParserContextFlags.Await);
+        }
+
+        function setAsyncParameterContext(val: boolean) {
+            setContextFlag(val, ParserContextFlags.AsyncParameter);

```

Comment:
I wish these were combined into a tri-state, ditto for Yield and GeneratorParameter.